object/src/mem.rs

```diff
@@ -618,34 +618,54 @@ where
     /// Insert a data element to the object, replacing (and returning) any
     /// previous element of the same attribute.
     pub fn put_element(&mut self, elt: InMemElement<D>) -> Option<InMemElement<D>> {
-        self.entries.insert(elt.tag(), elt)
+        self.entries.insert(elt.tag(), elt).map(|e| {
+            self.len = Length::UNDEFINED;
+            e
+        })
```

Comment:
Ah, so this is why I had a weird hunch about this. \*headscratch\* `BTreeMap::insert` only returns `Some` if the element already existed, but we want the length to be reset regardless of this.
Let's go with this instead.

```suggestion
        let o = self.entries.insert(elt.tag(), elt);
        self.len = Length::UNDEFINED;
        o
```